libs/MeshKernel/include/MeshKernel/BoundingBox.hpp

```diff
@@ -203,3 +208,31 @@ inline meshkernel::Vector meshkernel::BoundingBox::Delta() const
 {
     return Vector(m_upperRight.x - m_lowerLeft.x, m_upperRight.y - m_lowerLeft.y);
 }
+
+inline bool meshkernel::BoundingBox::Overlaps(const BoundingBox& other) const
+{
+
+    const auto& otherLowerleft = other.lowerLeft();
+    const auto& otherUpperRight = other.upperRight();
+    if (m_upperRight.x < otherLowerleft.x)
```

Comment:
optional: can have a single if statement  instead of 4 (use || between conditions)